src/common.rs

```diff
@@ -58,8 +58,8 @@ impl AppDataType {
     }
 }
 
-const ERR_NOT_SUPPORTED: &'static str = "App data directories not supported";
-const ERR_INVALID_APP_INFO: &'static str = "Invalid app name or author";
```

Comment:
This bumps the required rustc version from pre-1.12.0 to 1.17.0 - unless there are other compelling reasons for upping this requirement, do you think we could leave this as it is?

This lack of support for 1.16.0 is why `appveyor` failed.